src/nvim/os/fs.c

```diff
@@ -120,9 +120,13 @@ static bool is_executable(const char_u *name)
     return false;
   }
 
-  if (S_ISREG(mode) && (S_IXUSR & mode)) {
-    return true;
-  }
+#if WIN32
+  // Windows does not have exec bit, just check if
+  // the file exists
+  return (S_ISREG(mode));
```

Comment:
> Vim will still return 1 even though I can't run said Batch script from a Windows shell.

If we can make `executable()` do something actually useful on Windows, then it doesn't matter what Vim's behavior is. So `os_file_is_readable()` seems like a good idea to me.